src/dos/program_mount.cpp

```diff
@@ -374,7 +378,14 @@ void MOUNT::Run(void) {
 				delete Drives[drive_index(drive)];
 				Drives[drive_index(drive)] = nullptr;
 			} else {
-				newdrive = new localDrive(temp_line.c_str(),sizes[0],int8_tize,sizes[2],sizes[3],mediaid);
+				newdrive = new localDrive(
+				        temp_line.c_str(),
+				        sizes[0],
+				        int8_tize,
+				        sizes[2],
+				        sizes[3],
+				        mediaid,
+				        section->Get_bool("always_allow_rw_open"));
```

Comment:
The feature is all about letting users write-protect data files (or setting read-only protection), so I think the conf setting should use that kind of wording; ie: "readonly", "write-protected", etc, instead of "rw".

Reply:
You have to be mindful of var name lengths here because they affect column width in the config file.

Reply:
Yes, just thinking that we should avoid terms like "rw" and try to use "ro". 

Currently the users has to do some mental gymnastics to figure out the implication of "always_allow_rw_open" is thus to allow the use of write-protected or read-only files. 

(I unfortunately don't anything to offer as an alternative :sweat_smile: !)

Reply:
We use longer config names than this, and I think having descriptive and discoverable names trumps all other concerns. E.g we have `viewport_resolution`, not `viewport_res` for example, plus check out the mouse related names. The column width is not a problem in the grand scheme of things.

I agree with @kcgen on this, it should have one of the terms he listed in the name. I'll think about it.

Reply:
I think I like `allow_write_protected_files`; this name and @kcgen's revised description should make things clear. Thought?

Reply:
For users interested in this feature, I think that really helps, @johnnovak. 

If a user sets `allow_write_protected_files = false` but /has/ write-protected their games' data files, then this setting will make sense when these problematic games are denied access to these files.

Reply:
>  I think having descriptive and discoverable names trumps all other concerns

yup - especially settings like this, which are typically going to be change while the user is editing their conf text files.

(as opposed to on-the-fly command-line changes, like switching sbtype from sb16 to sbpro2).

Reply:
> > I think having descriptive and discoverable names trumps all other concerns
> 
> yup - especially settings like this. These are going to be "one-and-done", weighed against their personal workflow, while editing their conf text file.
> 
> As opposed to "tactical" changes on the command-line, like switching `sbtype` from `sb16` to `sbpro2` for some side-by-side comparisons.

Yeah I agree, and as one of my previous colleagues put it: as a developer, you're always optimising for _something_—whether consciously or out of sheer habit.

I'd very much prefer if we made these decisions consciously and based on things that really make a difference. For example, at the code level I'd say readability is of primary concern, and we should always optimise for that and easy maintainability. Due to other considerations (e.g. performance) we might relax that approach when necessary, but those special cases need to be well-justified and backed up with facts and numbers—otherwise we should default to well-structured code with descriptive variable and function names, etc.

Similarly for config parameters, I want all these parameter names to be as obvious as possible. `allow_write_protected_files` is 27 chars long, a bit longer than the other descriptive names we've added recently, but it describes the intent well. As a general rule, we could agree on keeping the names below 30-32 chars perhaps. 

For reference, these are some of the more recently added config params that all have good names and are not on th short side:

`mouse_middle_release`
`dos_mouse_immediate`
`ps2_mouse_model`
`sb_filter_always_on`
`ps1audio_dac_filter`
`viewport_resolution`
`pause_when_inactive`
`mcb_fault_strategy`

Reply:
Also, I'd like to point out that the description of the body text needs to be wrapped to 79 chars, indeed, but the length of the config params doesn't matter when you use the `config -h <paramname>` command:

<img width="1215" alt="image" src="https://user-images.githubusercontent.com/698770/219985715-320975c7-2b5c-4c54-9f17-c8bd6e729195.png">

When you view the autogenrated config file in an editor, the line-lengths are all over the place already anyway, and this is "unfixable", so I wouldn't worry about it at all. Just use a big enough window as shown on the screenshot, and problem solved. In the grand scheme of things, this is a very minor concern. I do think trying to enforce 78-80 columns in the generated config is pointless—even terminal users can simply enlarge their terminal window these days 😄 

<img width="1840" alt="image" src="https://user-images.githubusercontent.com/698770/219985853-82ee68d9-7de0-4116-a8b2-ce8e0a039492.png">

Reply:
All very well said @johnnovak !
This is great info that would fit well in the maintenance wiki page (or contributors docs!)